### ISSUE TITLE
Pi0 fast libero checkpoint no longer works with new repo?

### ISSUE THREAD

**Issue by HenryChen4** [Physical-Intelligence/openpi#849]
The only alterations I have made was the following:

```
  EnvMode.LIBERO: Checkpoint(
      config="pi0_fast_libero",
      dir="gs://openpi-assets/checkpoints/pi0_fast_libero",
  ),
```

and then subsequently ran python examples/libero/main.py. The arm just spins around with success rate of 0 for all tasks. Wondering what could have happened or if this checkpoint is no longer supported?

**Comment by BNU-IVC-YuanLIU** [comment]
I am experiencing the same issue. Could you please let me know how you resolved it?

**Comment by aojiaosaiban** [comment]
same issue here

**Comment by aopolin-lv** [comment]
Same issue. Further, I have finetune pi0-fast using the `pi0_fast_base` ckpt download from `gs://openpi-assets/checkpoints/pi0_fast_base` on the libero dataset and the success rate is still 0.

**Comment by ziyaosg** [comment]
Same issue here. The last commit that still achieves the expected success rate (~0.61 on libero_10) for me is `e4580662a60e528ee8e39e7073f82affec7dd407` from Sep 7, 2025.

**Comment by hzhhahah** [comment]
Same issue. I finetuned pi0 using the libero dataset from physical-intelligence/libero(on huggingface). When I evaluated the fine-tuned checkpoint on LIBERO-10, the robotic arm merely rotated. I compared the `norm_stats.json` file I generated myself against the one from `pi0.5-LIBERO`, and I noticed significant discrepancies between the mean and variance values ​​stored in the two files. However, theoretically speaking—assuming both were fine-tuned on the exact same dataset—these values ​​should be identical.

**Comment by Agenda8** [comment]
I’m running into a similar issue. When I evaluate the official pi0_libero checkpoint on LIBERO-10, the overall success rate is only around 60%, which is far below the 85.2% reported by the official benchmark.

After looking into it more closely, the main problem seems to be tasks 8 and 9 in LIBERO-10, where the success rate is nearly 0. In the failure videos, the model behaves normally for a while, but then suddenly starts moving erratically and spinning in place.

I’m not sure what is causing this, but it looks like the checkpoint may not be behaving as expected on these tasks. Has anyone else seen this issue, or is there a known reason for the degradation?

**Comment by 0xadvait** [comment]
Found the root cause. It is a normalization regression, and @ziyaosg's bisect to e4580662 was exactly right.

**What changed**

At e4580662, `LeRobotLiberoDataConfig` built its `DataConfig` without setting `use_quantile_norm`, so the pi0-FAST LIBERO configs trained and served with z-score normalization. On current main, `DataConfigFactory.create_base_config` force-sets `use_quantile_norm = model_type != PI0`, which silently flipped `pi0_fast_libero` (and `pi0_fast_libero_low_mem_finetune`) to quantile normalization. The released `pi0_fast_libero` checkpoint was trained with z-score stats, so on current main the state input is mis-normalized going in and, worse, the action chunk is mis-unnormalized coming out. That scale mismatch is why the arm spins instead of acting. DROID FAST configs are unaffected because they used quantile normalization both before and after the change.

You can confirm the flip without running a robot:

```python
import openpi.training.config as c
cfg = c.get_config("pi0_fast_libero")
print(cfg.data.create(cfg.assets_dirs, cfg.model).use_quantile_norm)
# e4580662: False        current main: True
```

**Empirical confirmation**

I ran the official `gs://openpi-assets/checkpoints/pi0_fast_libero` checkpoint on a fixed synthetic LIBERO observation at both commits (CPU, greedy FAST decoding, fully deterministic; two infer calls per commit return bitwise-identical actions):

- input transform sequences identical, preprocessed image hashes identical
- post-normalization state vector differs by up to 10.36 (e.g. dim 7: 37.50 vs 27.14)
- FAST token ids diverge at positions 27, 28, 30, 31, 32
- output action chunks differ with max abs diff 2.64 across the 10x7 chunk

So the same checkpoint and the same observation produce qualitatively different actions purely from the normalization flag.

**Fix**

#971 adds an explicit per-config override and pins the two `pi0_fast_libero` configs back to z-score normalization. With the fix applied, the probe outputs at current main match e4580662 again. Until it lands, the workaround is to check out e4580662 for pi0-FAST LIBERO evals, or apply the same one-flag change locally.

One caveat: anyone who finetuned pi0-FAST on LIBERO using current main trained with quantile normalization, and those checkpoints will want the current behavior; the override in the PR makes that a one-line config choice instead of a silent default.
